### DISCUSSION TITLE
Deprecated support for the 32bit OS Architecture

### DISCUSSION THREAD

**Discussion by Morsmalleo** [iBotPeaches/Apktool#3090] [Q&A] [answered]
G'day @iBotPeaches, I didn't feel right about opening this as an issue so I opened it as a question here.

I've been using *Apktool* for quite some time now and I love it, so first I want to say thank you for such a great work of art.

Second, I would like to ask about the deprecation of 32bit arch support, this morning when testing new updates for my project on a 32bit arch Kali Linux VM, everything worked fine until it came to running Apktool (currently utilising `apktool v2.6.1 snapshot`) to compile a decompiled APK. 

I am just wondering why the support for 32bit arch has dropped? I seen on some issues it had something to with aapt/aapt2 & `libwinp-something` (can't remember the name of it) if memory serves me correctly, but my expertise and understanding of Java, particularly with this project is quite limited unfortunately.

**Comment by iBotPeaches** [comment, marked as answer]
> I am just wondering why the support for 32bit arch has dropped?

Mainly because AOSP dropped it for build tools. So I was then mixing building 32/64 for platforms that could on top of aapt/aapt2 and even debating the introduction of arm support. So when I realized that modifying AOSP to build 32bit that it no longer did by default was consuming way too much time - it seemed easier to drop 32bit.

**Reply by Morsmalleo** [reply]
Ok that makes a lot of sense, sounded like a bit of a headache for you! So 64bit OS architecture is the go-to arch type for operating Apktool as of 2.7.0 is this correct if my understanding of what you have told me is correct also? 😁

**Reply by iBotPeaches** [reply]
Yeah 64bit has been suggested I think since like 2.5.0.

**Reply by Morsmalleo** [reply]
> Yeah 64bit has been suggested I think since like 2.5.0.

Awesome, thank you for answering my question very clearly, and thank you for the quick response to 😁
